authors/apps/comments/views.py

```diff
@@ -0,0 +1,143 @@
+        comment = request.data
+        comment.update({'author': request.user.pk, 'article': article.id})
+        serializer = serializers.CommentSerializer(data=comment)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response({"message": "Comment Successfully added"},
+                        status=status.HTTP_201_CREATED)
+
+    def get(self, *args, **kwargs):
+        slug = self.kwargs['slug']
+        try:
+            article = Article.objects.get(slug=slug)
+        except Article.DoesNotExist:
+            message = 'Article does not exist'
+            return Response(message, status.HTTP_404_NOT_FOUND)
+        comments = self.queryset.filter(article_id=article.id)
+        serializer = serializers.CommentSerializer(comments, many=True)
+        return Response({"comments": serializer.data,
+                         "commentsCount": comments.count()})
+
+
+class CommentApiView(generics.RetrieveUpdateDestroyAPIView):
+    permission_classes = (IsAuthenticated,)
+    serializer_class = serializers.CommentSerializer
+
+    def retrieve(self, request, id, *args, **kwargs):
+        slug = self.kwargs['slug']
+        try:
+            article = Article.objects.get(slug=slug)
+            try:
+                comment = Comment.objects.get(id=id)
+            except Comment.DoesNotExist:
+                message = 'Comment does not exist'
+                return Response(message, status.HTTP_404_NOT_FOUND)
+        except Article.DoesNotExist:
+            message = 'Article does not exist'
+            return Response(message, status.HTTP_404_NOT_FOUND)
+        serializer = serializers.CommentSerializer(comment)
+        return Response({"comment": serializer.data})
+
+    def update(self, request, id, *args, **kwargs):
+        slug = self.kwargs['slug']
+        try:
+            article = Article.objects.get(slug=slug)
+            try:
+                comment = Comment.objects.get(id=id)
+            except Comment.DoesNotExist:
+                message = 'Comment does not exist'
+                return Response(message, status.HTTP_404_NOT_FOUND)
+        except Article.DoesNotExist:
+            message = 'Article does not exist'
+            return Response(message, status.HTTP_404_NOT_FOUND)
+        Comment.objects.all().filter(pk=id).update(**request.data)
+        msg = "Comment successfully updated"
+        return Response({
+            "message": msg
+        }, status=status.HTTP_200_OK)
+
+    def destroy(self, request, id, *args, **kwargs):
+        slug = self.kwargs['slug']
+        try:
+            article = Article.objects.get(slug=slug)
+            try:
+                comment = Comment.objects.get(id=id)
+            except Comment.DoesNotExist:
+                message = 'Comment does not exist'
+                return Response(message, status.HTTP_404_NOT_FOUND)
+        except Article.DoesNotExist:
+            message = 'Article does not exist'
```

Comment:
@Opio-Emmanuel-Omona Return a dictionary here to maintain consistency through out the entire application.